### PR TITLE
Module8

### PR DESCRIPTION
**Pull request by ViktoriiaLozova**

### CODE REVIEW

File: src/main/java/module8/Passport.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    public int getNumber() {
+        return number;
+    }
+
+    public void setNumber(int number) {
+        this.number = number;
+    }
+
+    @Override
+    public boolean equals(Object o) {
```
Comment (AntonApanasovych) [inline src/main/java/module8/Passport.java:31]: Incorrect implementation for `equals`. You should also check for `null` and self reference.
I recommend to use IDEA equals and hashcode generator. Try combination `alt + insert`, displayed context menu should contain equals and hashcode generator.

File: src/main/java/module8/Passport.java [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+        Passport passport = (Passport) o;
+        return number.equals(passport.number) &&
+                series.equals(passport.series);
+    }
+
+    @Override
+    public int hashCode() {
```
Comment (AntonApanasovych) [inline src/main/java/module8/Passport.java]: hashcode is correct but it is good to use generator for both `equals` and `hashcode`.

File: src/main/java/module8/Task1.java [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    private static String decompress(String source) throws Exception {
+      String decodedChars = "";
+        for (int i = 0; i < source.length(); i++) {
+            char charFromstring = source.charAt(i);
+            if (charFromstring == '1' && !isNumber(source.charAt(i + 1)) && source.charAt(i - 1) != '1'
+                    || isNumber(charFromstring) && charFromstring != '1' && i == 0
+                    || charFromstring == '0' && source.charAt(i - 1) != '1') {
+                throw new Exception("Decoding error");
```
Comment (AntonApanasovych) [inline src/main/java/module8/Task1.java]: Do not use `Exception` for runtime errors. It is checked exception and compiler will force you to handle it in any method that will call your method. And also it is too broad.
More compliant solution to use `RuntimeException`. But it is also too broad, better to use concrete or even custom exception, i.e. IllegalArgumentException, IllegalStateException etc. If you will not be able to find concrete exception use `RuntimeException`

File: src/main/java/module8/Task3.java [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+Для поиска создайте новый экземпляр Passport в который внесите заведомо существующие данные (совпадающие с данными
+Passport на предыдущем шаге), проверьте поведение приложения если паспортные данные внесены с ошибкой.
+Обратите внимание на поведение HashMap в случае если Passport не имеет методов hashCode и equals
+*/
+
+import java.util.HashMap;
+
+public class Task3 {
+    public static void main(String[] args) {
+        HashMap<Passport, Person> map = new HashMap<>();
```
Comment (AntonApanasovych) [inline src/main/java/module8/Task3.java]: `Map<Passport, Person> map = new HashMap<>();`

File: src/main/java/module8/Passport.java (hunk 1 above)
Comment (ViktoriiaLozova) [reply src/main/java/module8/Passport.java:31]: Done. Hashcode and equals generator was used

File: src/main/java/module8/Task1.java (hunk 3 above)
Comment (ViktoriiaLozova) [reply src/main/java/module8/Task1.java]: Done. new Exception was changed to DecodingException

File: src/main/java/module8/Task3.java (hunk 4 above)
Comment (ViktoriiaLozova) [reply src/main/java/module8/Task3.java]: Done. HashMap was changed to Map

**Pull request merged**